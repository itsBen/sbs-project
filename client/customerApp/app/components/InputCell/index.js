import React, { Component } from 'react';
import { TextInput, View, StyleSheet } from 'react-native';

export default (InputCell = ({
  placeholder,
  maxLength,
  multiline,
  inputStyle,
  ...props
}) => (
  <View style={styles.container}>
    <TextInput
      style={[styles.inputField, inputStyle]}
      placeholder={placeholder}
      maxLength={maxLength}
      multiline={multiline}
      {...props}
    />
  </View>
));

const styles = StyleSheet.create({
  container: {
    paddingVertical: 18,
    paddingHorizontal: 15,
    flexDirection: 'row',
  },
  inputField: {
    fontSize: 16,
    flex: 1,
  },
});
