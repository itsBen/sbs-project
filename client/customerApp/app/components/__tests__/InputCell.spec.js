/* eslint-env node, jest */
import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import InputCell from '../InputCell';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <InputCell placeholder="Enter your password" secureTextEntry value="" />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
