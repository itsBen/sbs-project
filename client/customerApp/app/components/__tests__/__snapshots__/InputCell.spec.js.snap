// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "paddingHorizontal": 15,
      "paddingVertical": 18,
    }
  }
>
  <TextInput
    maxLength={undefined}
    multiline={undefined}
    placeholder="Enter your password"
    secureTextEntry={true}
    style={
      Array [
        Object {
          "flex": 1,
          "fontSize": 16,
        },
        undefined,
      ]
    }
    value=""
  />
</View>
`;
